### PR TITLE
Review Comments for release

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
… ifEmpty for reports

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by maxulysse — APPROVED** [review]

File: subworkflows/nf-core/gatk4/tumor_normal_somatic_variant_calling/main.nf [hunk 1]
```diff
@@ -121,15 +142,29 @@ workflow GATK_TUMOR_NORMAL_SOMATIC_VARIANT_CALLING {
...
                                     meta, cram, crai, intervals ->
 
-                                    [[patient:meta.patient, normal_id:meta.normal_id, tumor_id:meta.tumor_id, sex:meta.sex, id:meta.normal_id, num_intervals:meta.num_intervals],
+                                    [[
+                                        id:             meta.normal_id,
+                                        normal_id:      meta.normal_id,
+                                        num_intervals:  meta.num_intervals,
+                                        patient:        meta.patient,
+                                        tumor_id:       meta.tumor_id,
+                                        sex:            meta.sex,
```
Comment (SusiJo) [inline subworkflows/nf-core/gatk4/tumor_normal_somatic_variant_calling/main.nf:166]: Probably not a problem in a meta map, but all others have `tumor_id` as last elem

Comment (maxulysse) [reply subworkflows/nf-core/gatk4/tumor_normal_somatic_variant_calling/main.nf:166]: good point, we could sort all meta map alphabetically, that way, we don't have to think about it anymore

File: subworkflows/nf-core/variantcalling/manta/somatic/main.nf [hunk 2]
```diff
@@ -101,15 +144,27 @@ workflow RUN_MANTA_SOMATIC {
         MERGE_MANTA_SMALL_INDELS.out.vcf,
         manta_candidate_small_indels_vcf.no_intervals
     ).map{ meta, vcf ->
-        [[patient:meta.patient, normal_id:meta.normal_id, tumor_id:meta.tumor_id, sex:meta.sex, id:meta.tumor_id + "_vs_" + meta.normal_id],
+        [[
+            id:         meta.tumor_id + "_vs_" + meta.normal_id,
+            normal_id:  meta.normal_id,
+            patient:    meta.patient,
+            sex:        meta.sex,
+            tumor_id:   meta.tumor_id,
```
Comment (SusiJo) [inline subworkflows/nf-core/variantcalling/manta/somatic/main.nf:152]: Does the `variantcaller:"manta"` have to be added here as well?

File: subworkflows/nf-core/variantcalling/manta/tumoronly/main.nf [hunk 3]
```diff
@@ -48,7 +55,14 @@ workflow RUN_MANTA_TUMORONLY {
...
         manta_candidate_sv_vcf.intervals.map{ meta, vcf ->
 
-                [groupKey([patient:meta.patient, sample:meta.sample, status:meta.status, sex:meta.sex, id:meta.sample, num_intervals:meta.num_intervals],
+                [groupKey([
+                            id:             meta.sample,
+                            num_intervals:  meta.num_intervals,
+                            patient:        meta.patient,
+                            sample:         meta.sample,
+                            sex:            meta.sex,
+                            status:         meta.status
```
Comment (SusiJo) [inline subworkflows/nf-core/variantcalling/manta/tumoronly/main.nf:64]: Also wondering if the `variantcaller:"manta"` has to be added to al manta-related processes, but not sure 🤔

File: subworkflows/nf-core/gatk4/tumor_normal_somatic_variant_calling/main.nf (hunk 1 above)
Comment (FriederikeHanssen) [reply subworkflows/nf-core/gatk4/tumor_normal_somatic_variant_calling/main.nf:166]: tried doing that, clearly missed a few. I'll fix the ones you found now :)

File: subworkflows/nf-core/gatk4/tumor_only_somatic_variant_calling/main.nf [hunk 4]
```diff
@@ -64,7 +64,14 @@ workflow GATK_TUMOR_ONLY_SOMATIC_VARIANT_CALLING {
     MERGE_MUTECT2(
         mutect2_vcf_branch.intervals
         .map{ meta, vcf ->
-            new_meta = [patient:meta.patient, sample:meta.sample, status:meta.status, sex:meta.sex, id:meta.sample, num_intervals:meta.num_intervals]
+            new_meta = [
+                        id:             meta.sample,
+                        num_intervals:  meta.num_intervals,
+                        patient:        meta.patient,
+                        sample:         meta.sample,
+                        sex:            meta.sex,
```
[the remark this answers is not in the document]
Comment (FriederikeHanssen) [reply subworkflows/nf-core/gatk4/tumor_only_somatic_variant_calling/main.nf:72]: I am not sure I understand what the issue is here? It is sorted alphabetically

[the remark this answers is not in the document]
Comment (SusiJo) [reply subworkflows/nf-core/gatk4/tumor_only_somatic_variant_calling/main.nf:72]: ah right, then it's not sorted alphabetically in another map in that same file

File: subworkflows/nf-core/variantcalling/manta/somatic/main.nf (hunk 2 above)
Comment (FriederikeHanssen) [reply subworkflows/nf-core/variantcalling/manta/somatic/main.nf:152]: no these vcfs don't get annotated, strelka just needs them for bp computation. the meta field variantcaller is just used for writing the annotated vcfs to the proper output folders.

File: subworkflows/nf-core/variantcalling/manta/tumoronly/main.nf (hunk 3 above)
Comment (FriederikeHanssen) [reply subworkflows/nf-core/variantcalling/manta/tumoronly/main.nf:64]: It is added in the final formatting for the output. Here in these line the meta map is just adapted generate the groupKey for groupTuple on variable input

File: subworkflows/nf-core/gatk4/tumor_only_somatic_variant_calling/main.nf [hunk 5]
```diff
@@ -82,7 +89,14 @@ workflow GATK_TUMOR_ONLY_SOMATIC_VARIANT_CALLING {
     MERGEMUTECTSTATS(
         mutect2_stats_branch.intervals
         .map{ meta, stats ->
-            new_meta = [patient:meta.patient, sample:meta.sample, status:meta.status, sex:meta.sex, id:meta.sample, num_intervals:meta.num_intervals]
+            new_meta = [
+                        id:             meta.sample,
+                        num_intervals:  meta.num_intervals,
+                        patient:        meta.patient,
+                        sample:         meta.sample,
+                        status:         meta.status,
```
Comment (SusiJo) [inline subworkflows/nf-core/gatk4/tumor_only_somatic_variant_calling/main.nf:97]: switch `status` and `sex`

File: subworkflows/nf-core/gatk4/tumor_only_somatic_variant_calling/main.nf [hunk 6]
```diff
@@ -98,7 +112,14 @@ workflow GATK_TUMOR_ONLY_SOMATIC_VARIANT_CALLING {
...
             mutect2_f1r2_branch.intervals
             .map{ meta, f1r2 ->
-                new_meta = [patient:meta.patient, sample:meta.sample, status:meta.status, sex:meta.sex, id:meta.sample, num_intervals:meta.num_intervals]
+                new_meta = [
+                            id:             meta.sample,
+                            num_intervals:  meta.num_intervals,
+                            patient:        meta.patient,
+                            sample:         meta.sample,
+                            status:         meta.status,
+                            sex:            meta.sex,
```
Comment (SusiJo) [inline subworkflows/nf-core/gatk4/tumor_only_somatic_variant_calling/main.nf:121]: switch here as well

File: subworkflows/nf-core/variantcalling/tiddit/main.nf [hunk 7]
```diff
@@ -20,8 +20,24 @@ workflow RUN_TIDDIT {
     tiddit_ploidy = TIDDIT_SV.out.ploidy
     tiddit_vcf_gz = TABIX_BGZIP_TIDDIT_SV.out.gz_tbi.map{ meta, gz, tbi ->
 
-        new_meta = meta.tumor_id ? [patient:meta.patient, normal_id:meta.normal_id, tumor_id:meta.tumor_id, sex:meta.sex, id:meta.tumor_id + "_vs_" + meta.normal_id, num_intervals:meta.num_intervals, variantcaller:'tiddit']
-                                        : [patient:meta.patient, sample:meta.sample, status:meta.status, sex:meta.sex, id:meta.sample, num_intervals:meta.num_intervals, variantcaller:'tiddit']
+        new_meta = meta.tumor_id ? [
+                                        id:             meta.tumor_id + "_vs_" + meta.normal_id,
+                                        patient:        meta.patient,
```
Comment (SusiJo) [inline subworkflows/nf-core/variantcalling/tiddit/main.nf:25]: put `patient`after `num_intervals`

Comment (SusiJo) [reply subworkflows/nf-core/variantcalling/tiddit/main.nf:25]: also: is `sample` missing here maybe?

File: subworkflows/nf-core/gatk4/tumor_only_somatic_variant_calling/main.nf (hunk 4 above)
[the remark this answers is not in the document]
Comment (FriederikeHanssen) [reply subworkflows/nf-core/gatk4/tumor_only_somatic_variant_calling/main.nf:72]: found it

**Review by maxulysse — APPROVED** [review]

File: workflows/sarek.nf [hunk 8]
```diff
@@ -660,7 +709,7 @@ workflow SAREK {
                 ch_table_bqsr_no_spark,
                 ch_table_bqsr_spark)
 
-            ch_reports  = ch_reports.mix(ch_table_bqsr.map{ meta, table -> table})
+            ch_reports  = ch_reports.mix(ch_table_bqsr.collect{ meta, table -> table})
```
Comment (SusiJo) [inline workflows/sarek.nf:712]: Can the `collect` replace the `map`? Or should it be both?

**Pull request merged**